core/domain/user_jobs_one_off_test.py

```diff
@@ -196,6 +196,63 @@ def test_user_contributions_get_created_after_running_the_job(self):
             ['exp_id'])
 
 
+class UserAuthModelOneOffJobTests(test_utils.GenericTestBase):
```

Comment:
What about a test case for running the migration job multiple times?

Reply:
Please see the updated tests now. As mentioned in PR description, this migration job is only supposed to be run once. I have written tests `test_one_off_job_gae_id_registered_for_another_user_raises_error` and `test_one_off_job_profile_registered_with_another_gae_id_raises_error` - there I have run migration jobs twice to check if exception is raised correctly or not. PTAL and tell if that fulfils the need.

Reply:
Note that map reduce jobs can fail midway due to unexpected reasons. You need to make sure your job is resilient to this--it should be able to be restarted and finish the migration successfully even if some data was already migrated. Migration jobs should also be idempotent--if they iterate over migrated data, that data shouldn't change no matter how many subsequent runs occur.